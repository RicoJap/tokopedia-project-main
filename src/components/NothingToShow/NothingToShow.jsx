import React from "react";
import Image from "../Image/Image";
import Typography from "../Typography/Typography";

import Pokeball from "../../assets/pokeball-big.png";

const styles = {
  image: {
    height: 100,
    width: 100,
    display: "block",
    margin: "auto",
    paddingTop: "50%",
  },
  label: {
    display: "block",
    textAlign: "center",
    margin: "1em",
  },
};

const NothingToShow = ({ label }) => {
  return (
    <div>
      <Image src={Pokeball} variant="square" style={styles.image} />
      <Typography
        variant="p"
        color="textSecondary"
        label={label}
        style={styles.label}
      />
    </div>
  );
};

export default NothingToShow;
