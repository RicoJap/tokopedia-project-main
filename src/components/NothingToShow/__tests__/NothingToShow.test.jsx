import "@testing-library/jest-dom";
import React from "react";
import renderer from "react-test-renderer";

import { fireEvent, render, screen, waitFor } from "@testing-library/react";

import NothingToShow from "../NothingToShow";

const MockNothingToShow = () => {
  return <NothingToShow label={'Test Label'} />;
};

describe("Snapshot testing", () => {
  const tree = renderer.create(<MockNothingToShow />).toJSON();
  expect(tree).toMatchSnapshot();
});

describe("NothingToShow Component renders correctly", () => {
  test("Check if NothingToShow component is present in the document", () => {
    render(<MockNothingToShow />);
    expect(screen.getByText("Test Label")).toBeInTheDocument();
    expect(screen.getByTestId("image")).toBeInTheDocument();
  });
});
