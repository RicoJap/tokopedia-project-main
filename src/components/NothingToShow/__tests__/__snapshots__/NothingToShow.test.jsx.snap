// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<div
  data-testid="nothing-to-show"
>
  <div
    className="MuiAvatar-root MuiAvatar-square"
    data-testid="image"
    style={
      Object {
        "display": "block",
        "height": 100,
        "margin": "auto",
        "paddingTop": "50%",
        "width": 100,
      }
    }
  >
    <img
      className="MuiAvatar-img"
      src="pokeball-big.png"
    />
  </div>
  <span
    className="MuiTypography-root MuiTypography-colorTextSecondary"
    style={
      Object {
        "display": "block",
        "margin": "1em",
        "textAlign": "center",
      }
    }
  >
    Test Label
  </span>
</div>
`;
